src/validationinterface.h

```diff
@@ -42,6 +42,16 @@ void UnregisterAllValidationInterfaces();
  * will result in a deadlock (that DEBUG_LOCKORDER will miss).
  */
 void CallFunctionInValidationInterfaceQueue(std::function<void ()> func);
+/**
+ * This is a synonym for the following, which asserts certain locks are not
+ * held:
+ *     std::promise<void> promise;
+ *     CallFunctionInValidationInterfaceQueue([&promise] {
+ *         promise.set_value();
+ *     });
+ *     promise.get_future().wait();
+ */
+void BlockOnValidationInterfaceQueueDrain();
```

Comment:
If I understand this correctly, it's not actually waiting until a drain, as new events can be added to the queue before the callback added in this function is executed.

It's more a synchronization point.